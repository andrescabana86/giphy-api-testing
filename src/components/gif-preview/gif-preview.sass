.gif-preview
  align-items: center
  border: 1px solid gray
  display: flex
  flex-direction: column
  justify-content: center
  padding: 4px
  &__image
    display: block
    height: 72px
    margin-bottom: 4px
    width: 72px
    &-src
      height: 100%
      width: 100%