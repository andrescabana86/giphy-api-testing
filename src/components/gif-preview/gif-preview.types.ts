export type TGifPreviewProps = {
  selected: boolean
  url: string
}
