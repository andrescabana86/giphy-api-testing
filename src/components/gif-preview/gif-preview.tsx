import React from 'react'
import { LikeButton } from '@components/like-button/like-button'
import { TGifPreviewProps } from './gif-preview.types'
import './gif-preview.sass'

export const GifPreview = React.memo((props: TGifPreviewProps) => {
  const { url, selected } = props
  return (
    <section className="gif-preview" data-testid="gif-preview">
      <picture className="gif-preview__image">
        <img
          alt="Like"
          className="gif-preview__image-src"
          data-testid="gif-preview__image-src"
          src={url}
        />
      </picture>

      <LikeButton active={selected} />
    </section>
  )
})
