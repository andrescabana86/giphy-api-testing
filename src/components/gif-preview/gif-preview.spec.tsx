import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { GifPreview } from './gif-preview'

describe('GifPreview', () => {
  afterEach(cleanup)

  it('should render GifPreview component', () => {
    // Arrange
    const id = 'uniqueID'
    const url = '/this/is/fake/url'

    // Act
    const { asFragment } = render(<GifPreview id={id} url={url} selected={false} />)

    // Assert
    expect(asFragment()).toMatchSnapshot()
  })

  it('should has "fake url" when it is rendered properly"', () => {
    // Arrange
    const id = 'uniqueID'
    const url = '/this/is/fake/url'

    // Act
    const { getByTestId } = render(<GifPreview id={id} url={url} selected={false} />)
    const component = getByTestId('gif-preview')
    const image = getByTestId('gif-preview__image-src')

    // Assert
    expect(component).toHaveClass('gif-preview')
    expect(image).toHaveClass('gif-preview__image-src')
    expect(image).toHaveAttribute('src', url)
  })

  it('should has LikeButton when it is rendered properly"', () => {
    // Arrange
    const id = 'uniqueID'
    const url = '/this/is/fake/url'

    // Act
    const { getByTestId } = render(<GifPreview id={id} url={url} selected={false} />)
    const button = getByTestId('like-button')

    // Assert
    expect(button).not.toEqual(null)
  })
})
