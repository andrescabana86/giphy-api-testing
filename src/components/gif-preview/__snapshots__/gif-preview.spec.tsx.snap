// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifPreview should render GifPreview component 1`] = `
<DocumentFragment>
  <section
    class="gif-preview"
    data-testid="gif-preview"
  >
    <picture
      class="gif-preview__image"
    >
      <img
        alt="Like"
        class="gif-preview__image-src"
        data-testid="gif-preview__image-src"
        src="/this/is/fake/url"
      />
    </picture>
    <button
      class="like-button"
      data-testid="like-button"
    >
      Like
      <img
        alt="Like"
        class="like-button__icon"
        src="static/icons/heart-empty-red.svg"
      />
    </button>
  </section>
</DocumentFragment>
`;
